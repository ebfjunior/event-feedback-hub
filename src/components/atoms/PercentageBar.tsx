import React from 'react';

export type PercentageBarProps = {
  value?: number | null;
  className?: string;
};

export function PercentageBar({ value, className }: PercentageBarProps) {
  const clamped = Math.min(Math.max(value ?? 0, 0), 100);
  return (
    <div className={className}>
      <div className="h-2 w-full overflow-hidden rounded-full bg-muted">
        <div className="h-full rounded-full bg-emerald-500 transition-all" style={{ width: `${clamped}%` }} />
      </div>
    </div>
  );
}


