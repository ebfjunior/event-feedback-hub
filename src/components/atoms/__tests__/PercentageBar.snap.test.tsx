import React from 'react';
import { render } from '@testing-library/react';
import { PercentageBar } from '../PercentageBar';

describe('PercentageBar (snapshot)', () => {
	it('renders clamped values', () => {
		const { container: c1 } = render(<PercentageBar value={-10} />);
		expect(c1.firstChild).toMatchSnapshot();
		const { container: c2 } = render(<PercentageBar value={50} />);
		expect(c2.firstChild).toMatchSnapshot();
		const { container: c3 } = render(<PercentageBar value={110} />);
		expect(c3.firstChild).toMatchSnapshot();
	});
});


