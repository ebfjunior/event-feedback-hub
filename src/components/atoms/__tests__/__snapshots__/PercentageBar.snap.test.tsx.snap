// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PercentageBar (snapshot) > renders clamped values 1`] = `
<div>
  <div
    class="h-2 w-full overflow-hidden rounded-full bg-muted"
  >
    <div
      class="h-full rounded-full bg-emerald-500 transition-all"
      style="width: 0%;"
    />
  </div>
</div>
`;

exports[`PercentageBar (snapshot) > renders clamped values 2`] = `
<div>
  <div
    class="h-2 w-full overflow-hidden rounded-full bg-muted"
  >
    <div
      class="h-full rounded-full bg-emerald-500 transition-all"
      style="width: 50%;"
    />
  </div>
</div>
`;

exports[`PercentageBar (snapshot) > renders clamped values 3`] = `
<div>
  <div
    class="h-2 w-full overflow-hidden rounded-full bg-muted"
  >
    <div
      class="h-full rounded-full bg-emerald-500 transition-all"
      style="width: 100%;"
    />
  </div>
</div>
`;
